### ISSUE TITLE
Enemies appearing black above grates in DoTM.

### ISSUE THREAD

**Issue by adelpha** [Shpoike/Quakespasm#69]
Enemies are appearing black above certain grates in DoTM, as if lighting information isn't being picked up, or is being picked up from the wrong surface.

https://youtu.be/lrmQSEon24k

![Thanks Garoto for the screenshot](https://user-images.githubusercontent.com/11768175/132759166-bdd5ff89-e05e-4833-a5a8-af25705ab1a1.png)

.
This could be a red herring, but I've noticed that the grate models' interior surfaces are not culled and are 100% black.
.

![spasm0001](https://user-images.githubusercontent.com/13021896/132769044-63368fc7-b08f-4727-81f5-03a86eef7250.jpg)
![spasm0002](https://user-images.githubusercontent.com/13021896/132769058-998db3cc-b047-4b7e-b624-bb1596dc9b04.jpg)

**Comment by NightFright2k19** [comment]
Does it still happen with latest build from Sep 13?

**Comment by adelpha** [comment]
Yep, still happening.

FYI, vkQuake also suffers from this bug. Are these grates and stairs actually models rather than built from brushes?

**Comment by garoto** [comment]
Potential same issue being discussed here: https://github.com/sezero/quakespasm/issues/7

**Comment by adelpha** [comment]
Yeah, that's the same issue for sure. I mean, technically this could probably be considered a content bug that Kex just doesn't expose because it seems to use different logic for tracing to the nearest lit surface below.

OTOH, since their implementation works in DoTM content and I haven't noticed any glaring issues lighting issues in any vanilla or MP content in Kex, their implementation might be "better".

Total guess: Kex is tracing to the lit top surface of the grates, while QS/QSS/VK are all tracing to the unlit surface _inside_ the grates.

**Comment by garoto** [comment]
Personally, I don't give two shits about this `Kex engine` whatsoever. I hope if goes into obscurancy as fast as it came to be.
